### PR TITLE
Refactor learning rate scheduler

### LINKED ISSUE #46: Take another look at learning rate schedulers
Is there a better way of doing this? It should just be a mathematical function of the number of iterations.

### PR DESCRIPTION
**Pull request by samedii**
Implements #46 

Usage:

    LearningRateScheduler(
        optimizer,
        config['learning_rate'],
        starcompose(
            warmup(100),
            decay(0.97),
        ),
    ).attach(trainer)

Can easily be changed to using `partial(warmup, n_steps=100)` or classes `Warmup(100)` with `__call__`. I prefer partial or the current implementation.

### CODE REVIEW

**Comment by FelixAbrahamsson** [comment]
Nice! One question though, in the implementation of warmup it looks to me that the learning rate will be a constant 0 for `n_steps`. Shouldn't it be a linear increase up to `multiplier = 1`?

**Comment by samedii** [comment]
I think warmup of 0 is really good in transfer learning for gathering batch statistics and better gradient statistics before starting but let's check this!

File: workflow/ignite/handlers/learning_rate/warmup.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+
+def warmup(n_steps):
+    def _warmup(step, multiplier):
```
Comment (fippli) [inline workflow/ignite/handlers/learning_rate/warmup.py:4]: I really like this naming approach since there is no good way of returning anonymous functions in python

Comment (FelixAbrahamsson) [reply workflow/ignite/handlers/learning_rate/warmup.py:4]: A good reason not to name the wrapped function something generic like "wrapped_fn".

**Comment by FelixAbrahamsson** [comment]
> I think warmup of 0 is really good in transfer learning for gathering batch statistics and better gradient statistics before starting but let's check this!

That's interesting! I think the common implementation increases it linearly, but we could just have 2 different implementations. One with linear increase and one that keeps it at constant 0.

**Comment by samedii** [comment]
After digging a bit and thinking we should have linear warmup as default. It makes sense that the gradient statistics are affected by the step and only gathering statistics without any steps will skew them. You can always do "zero_warmup" before starting normal warmup though :) I will change this

**Comment by samedii** [comment]
Fully tested on utter

**Review by FelixAbrahamsson — APPROVED** [review]
